tests/conftest.py

```diff
@@ -15,6 +15,12 @@
 @pytest.fixture(autouse=True)
 def git_sandbox(monkeypatch: pytest.MonkeyPatch, tmp_path: Path):
     """Ensure git commands are executed without the current user settings"""
+    # Clear any GIT_ prefixed environment variable
+    for var in os.environ:
+        if var.startswith("GIT_"):
+            monkeypatch.delenv(var)
```

Comment:
Will it affect user's environment?

Reply:
No, it is only for the tests using the fixture (all given the fixture has `autouse=True`. 
It isolate from user-defined `GIT_` variables which are interfering with test, especially in `pre-commit` where some of those are set by `pre-commit` .
The `monkeypatch` fixture ensures that:
- this is contained to the current test and does not leak on other tests nor to the user environement
- this is properly cleaned at the end of the test